presto-main/src/main/java/io/prestosql/cost/PlanNodeLocalCostEstimate.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.cost;
+
+import io.prestosql.sql.planner.plan.PlanNode;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.Double.NaN;
+
+public class PlanNodeLocalCostEstimate
```

Comment:
Can we keep `PlanNodeCostEstimate` for the moment?
For the purpose of peak memory estimation I think we can simply add `peakMemoryWhenOutputting` to `PlanNodeCostEstimate`.
Note that separating classes for cumulative and local costs doesn't help with peak memory calculations as `peakMemoryWhenOutputting/peakMemory` are features of both cumulative and local costs.
I would prefer to limit the scope of PR to unifying approach of cost/stats provider (that is making cost calculator to be cumulative) and make actual separation in subsequent PRs.